R/data.table.R

```diff
@@ -1739,22 +1739,48 @@ replace_dot_alias = function(e) {
         GForce = FALSE
       } else {
         # Apply GForce
+        noCall_noVars = function(expr) (!is.call(expr) || length(all.vars(expr, max.names=1L, unique=FALSE))==0L) && !dotN(expr)
+        .gshift_ok = function(q) {
+          q = match.call(shift, q)
+          noCall_noVars(q[["n"]]) &&
+            noCall_noVars(q[["fill"]]) &&
+            noCall_noVars(q[["type"]]) &&
+            is.null(q[["give.names"]])
+        }
+        .ghead_ok = function(q) {
+          length(q) == 3L &&
+            length(q[[3L]]) == 1L &&
+            noCall_noVars(q[[3L]])
+        }
+        ".g[_ok" = function(q, x) {
+          length(q) == 3L &&
+            length(q[[3L]]) == 1L &&
+            noCall_noVars(q[[3L]]) &&
+            (q[[1L]] == "[" || (q[[1L]] == "[[" && eval(call('is.atomic', q[[2L]]), envir=x))) &&
```

Comment:
maybe simpler since we already know `q[[1L]] %in% c("[", "[[")`:
```suggestion
            (q[[1L]] != "[[" || eval(call('is.atomic', q[[2L]]), envir=x))) &&
```